Add GATK, fix stats accumulation for multiple samples

entee opened the pull request:
Most changes here made by Lucy

entee committed:
Add GATK, fix stats accumulation for multiple samples

Most changes here made by Lucy

entee commented on process_seq.py:
```diff
@@ -416,16 +442,26 @@ def extract_picard_stats(path):
-    writer = DictWriter(statsout, final_stats[0].keys(), delimiter="\t")
-    writer.writeheader()
-    writer.writerows(final_stats)
+    # Extract keys (Metric names, e.g. number of snps) from first dictionary
```
I'm having a little trouble understanding what's going on here and why switch to the raw csv writer instead of `DictWriter`. `DictWriter` is preferred because 1.) the code is more compact 2.) It's more resilient to changes, and handles column names cleanly (unsure if as written this does)

lucymaybio replied:
i have no problem using DictWriter, i just couldnt figure out how to format it so that the final-stats.csv had metrics as 1st column, and samples occupying each column therafter (i.e. swapping the cols for rows in your previous code - where the metrics were the first row, and samples occupied each row thereafter)

lucymaybio replied:
is this not very pythonic?

entee commented on process_seq.py:
```diff
@@ -317,8 +345,9 @@ def extract_bcf_stats(path):
                          "number of multiallelic SNP sites:"}
-
-    stats = {}
+    p = str(path.resolve()).split("/")[-1]
```
This path manipulation seems hacky and should probably be done in `configure_paths`. Either way, I can't tell what this is doing, it's just pulling parts of a split string so is likely quite brittle. It looks like this is pulling in a path, why can't this path be described in `configure_paths`?

lucymaybio replied:
there is definitely a better way of doing this - i just basically wanted to add another entry to the stats dict, so that when writing to csv for final-stats.csv the first row of the sample-specific column was the sample name (see the resulting outputted final-stats.csv from this code)

entee reviewed (commented):
I also need you to send me the GATK jar file you used, it looks like GATK's docker files are more involved than the one we've built.

lucymaybio committed:
Update2 process_seq.py

Hey nicolas, not sure if you got my previous update regarding the 3rd argument input (allele fraction, AF) - so here it is. I also tried to edit/propose changes to the README to add the AF input requirement. Hopefully that worked/you are able to see this...? Thanks!

lucymaybio committed:
add samtools flagstat, tabix path, vcf file path

Hey nicolas, 

sorry to keep pestering you with updates, but i realized i made a few mistakes in the previous updates.
 
1. I forgot to mention that another tool the updated script uses is tabix (used for indexing files), so we need to ensure that tabix is built into the docker container, and that the right tool path is used (also need to do this for GATK)

2. somehow i forgot to include the file path for the vcf file (very essential), so i added that too

3. i also collected more stats via 'samtools flagstat' command, so i've updated script to include this command and the output txt file. 

(previous updates = adding AF as input --> i accidentally used wrong syntax so updated that (needed to say args.AF, not just AF).

Other than that, i think its good to go! Thanks so much!!!!!!

entee committed:
Merge pull request #2 from lucymaybio/patch-1

add 3rd input argument, AF - allele fraction

entee committed:
Merge branch 'add_gatk' into patch-2

entee committed:
Merge pull request #3 from lucymaybio/patch-2

add samtools flagstat, tabix path, vcf file path

entee committed:
Small formatting changes, absorb Lucy PRs

entee committed:
Fix some of the paths, first pass rebuild Docker image

entee committed:
Baseline functioning with some stats confirmed on new demo data

entee committed:
Fix GATK path issue

entee committed:
Rework sample name stats setting from path

entee committed:
Convert to pandas for output of final stats

entee committed:
Small changes to readme, add requirements file

entee committed:
Final setup and cleanup

- Added new sample data
- Further clarifications in readme
- Parametrize GATK command
- Bump Docker image version to 0.3

entee committed:
Small modification to remove unnecessary data in docker container

Clarification comment in docker file
